### PR TITLE
Local Storage Range Allow 0:0

### LINKED ISSUE #6115: [Bug]: local storage - adapter errors in GetRange() when index range is [0:0]
### What happened?

Current Behavior:

While working on Spark/Iceberg/lakeFS and using local storage, the iceberg client perform an HTTP request for GetObject to the S3 Gateway with the range `Range: bytes=0-0`. 

lakeFS server calls [BlockStore.GetRange](https://github.com/treeverse/lakeFS/blob/master/pkg/gateway/operations/getobject.go#L103) calls [GetRange](https://github.com/treeverse/lakeFS/blob/master/pkg/block/local/adapter.go#L329) and returns an error with invalid index. 
 
When using S3 blockstore it works, when looking at the [HTTP specs seems like this should be supported](https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests#single_part_ranges).

Steps to Reproduce:
1. call getObject with s3 gateway using local storage with range 0-0

### Expected Behavior

This should return the first byte. 

### lakeFS Version

0.103.0

### Deplyoment

local

### Relevant logs output

```shell
DEBUG  [2023-06-18T17:04:02+03:00]opt/homebrew/Cellar/go/1.19.5/libexec/src/net/http/server.go:2109 net/http.HandlerFunc.ServeHTTP HTTP call ended                               client="Hadoop 3.3.5, aws-sdk-java/1.12.316 Mac_OS_X/12.5.1 OpenJDK_64-Bit_Server_VM/11.0.17+8-LTS java/11.0.17 scala/2.13.6 vendor/Azul_Systems,_Inc. cfg/retry-mode/legacy" host="localhost:8000" log_audit=true matched_host=false method=GET operation_id=get_object path=db/mytable/metadata/version-hint.text ref=main repository=example-repo request_id=9972ae57-4301-4d92-a960-52463e5ba0b0 sent_bytes=391 service_name=s3_gateway source_ip="127.0.0.1:49651" status_code=500 took=27.50475ms user=admin
```

### PR DESCRIPTION
**Pull request by Isan-Rivkin**
### Linked Issue

Fixes #6115

### CODE REVIEW

File: pkg/block/local/adapter.go [hunk 1]
```diff
@@ -325,7 +325,7 @@ func (l *Adapter) Exists(_ context.Context, obj block.ObjectPointer) (bool, erro
 }
 
 func (l *Adapter) GetRange(_ context.Context, obj block.ObjectPointer, start int64, end int64) (io.ReadCloser, error) {
-	if start < 0 || end <= 0 {
+	if start < 0 || end < 0 {
```
Comment (N-o-Z) [inline pkg/block/local/adapter.go:328]:
```suggestion
	if start < 0 || end < start {
```

Comment (N-o-Z) [reply pkg/block/local/adapter.go:328]: Consider this change

**Review by N-o-Z — CHANGES_REQUESTED** [review]
Change suggestion

File: pkg/block/local/adapter.go (hunk 1 above)
Comment (Isan-Rivkin) [reply pkg/block/local/adapter.go:328]: like +1

**Review by N-o-Z — APPROVED** [review]
Thanks !

**Review by nopcoder — APPROVED** [review]

**Pull request merged**